addons/mrp/models/mrp_workorder.py

```diff
@@ -553,7 +553,11 @@ def _plan_workorder(self, replan=False):
     def _cal_cost(self):
         total = 0
         for wo in self:
-            duration = sum(wo.time_ids.mapped('duration'))
+            intervals = list()
+            for time in wo.time_ids:
+                intervals.append((time.date_start, time.date_end))
+
+            duration = get_duration_from_intervals(intervals)
```

Comment:
Doesn't the  [`Intervals`](https://github.com/odoo/odoo/blob/17.0/addons/resource/models/utils.py#L112) class sufficient ? 
```suggestion
            ints = Intervals([[t.date_start, t.date_end, t] for t in wo.time_ids])
            duration = sum_intervals(ints) * 60
```